plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -1768,6 +1769,35 @@ def build_update_cluster_args(options, args)
     end
 
     def update_cluster(gear, options)
+      # NOTE: There is a check in place for head-gears to ensure a head-gear can not exceed
+      # the quota buffer, 120% of original quota.
+      # It is possible here that a non-primary gear can bump a head-gear's quota by more than the
+      # above buffer.  However, if this happens, any start, restart or move of a head-gear that is
+      # over quota limit AND over quota buffer limit will not succeed.
+      app = gear.application
+      head_gear_quota = get_quota(app)
+      # set head_gear quotas if necessary
+      # will be necessary if trying to move a non-primary gear before bumping
+      # an over-the-quota-limit head gear
+      head_blocks_current_limit = Integer(head_gear_quota[3])
+      head_blocks_used = Integer(head_gear_quota[1])
+      head_blocks_increment = head_blocks_current_limit * 1.01
+      head_quota_blocks_buffer = head_blocks_current_limit * 1.20
+
```

Comment:
these hard-coded numbers are fine with me, but I don't know whether they will be okay for an OSE customer.  you should probably run that by someone more familiar with OSE deployments; they might want something configurable.